src/Microsoft.TestPlatform.Utilities/InferRunSettingsHelper.cs

```diff
@@ -516,7 +516,11 @@ private static void VerifyCompatibilityWithOSArchitecture(Architecture architect
         {
             var osArchitecture = XmlRunSettingsUtilities.OSArchitecture;
 
-            if (architecture == Architecture.X86 && osArchitecture == Architecture.X64)
+            if (architecture == Architecture.X86 && (osArchitecture == Architecture.X64 || osArchitecture == Architecture.ARM64))
+            {
+                return;
+            }
+            if ((architecture == Architecture.ARM || architecture == Architecture.X86) && osArchitecture == Architecture.ARM64)
```

Comment:
The combination `architecture == Architecture.X86 && osArchitecture == Architecture.ARM64` is checked twice. Using `IsWow64GuestMachineSupported` may be more appropriate here.